src/background_thread.c

```diff
@@ -116,24 +116,28 @@ set_current_thread_affinity(int cpu) {
 #if defined(JEMALLOC_HAVE_SCHED_SETAFFINITY) || defined(JEMALLOC_HAVE_PTHREAD_SETAFFINITY_NP)
 #if defined(JEMALLOC_HAVE_SCHED_SETAFFINITY)
 	cpu_set_t cpuset;
-#else
+#elif defined(JEMALLOC_HAVE_PTHREAD_GETAFFINITY_NP)
```

Comment:
Question: does Hurd have pthread_getaffinity_np but not pthread_setaffinity_np? Asking because L116 have the `defined(JEMALLOC_HAVE_PTHREAD_SETAFFINITY_NP)` check.

This section of the code is assuming background thread is supported, so we should not have unsupported cases workaround here. See L86 around `#ifndef JEMALLOC_BACKGROUND_THREAD`. If it's not possible to support bg thds on hurd, let's make it explicitly so. Maybe through: https://github.com/jemalloc/jemalloc/blob/72cfdce71806443f4ccdbfe10aa5d50346a3d07e/configure.ac#L2575

Reply:
> does Hurd have pthread_getaffinity_np but not pthread_setaffinity_np?

No, it has none of the two (and not sched_setaffinity either)

> This section of the code is assuming background thread is supported

Mmm, ok, but is thread binding really necessary for background thread support?